### PR TITLE
[receiver/vcenter]Fix nil pointer exception when vcenter receiver collectVMs.

### PR DESCRIPTION
**Pull request by wzy531a**
**Description:** <Describe what has changed.>
Fix nil pointer exception when vcenter receiver collectVMs.

**Link to tracking Issue:** <Issue number if applicable>
[PR#16277](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16277)

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/vcenterreceiver/scraper.go [hunk 1]
```diff
@@ -269,6 +269,9 @@ func (v *vcenterMetricScraper) collectVMs(
 			continue
 		}
 
+		if moVM.Config == nil {
+			continue
```
Comment (djaglowski) [inline receiver/vcenterreceiver/scraper.go:273]: Is this an unexpected condition? Should we add an error to `errs`?

Comment (wzy531a) [reply receiver/vcenterreceiver/scraper.go:273]: At present, I am not sure what caused this situation. My guess is that the virtual machine was disconnected during the collection process, but such a problem should not cause the process to stop.
What kind of error message should I add?

Comment (djaglowski) [reply receiver/vcenterreceiver/scraper.go:273]: I think just something like `errs.AddPartial(1, fmt.Errorf("vm config empty for %s", <some id if possible>))

**Review by djaglowski — APPROVED** [review]

**Comment by djaglowski** [comment]
LGTM, but there is a lint failure

**Pull request merged**